src/python/pants/engine/internals/selectors.py

```diff
@@ -72,58 +70,57 @@ def parse_product_and_subject_types(
                 f"Expected either two or three arguments, but got {len(get_args)} arguments."
             )
 
-        product_expr = get_args[0]
-        if not isinstance(product_expr, ast.Name):
+        output_expr = get_args[0]
+        if not isinstance(output_expr, ast.Name):
             raise parse_error(
-                "The first argument should be the type of the product, like `Digest` or "
-                "`ProcessResult`."
+                "The first argument should be the output type, like `Digest` or `ProcessResult`."
             )
-        product_type = product_expr.id
+        output_type = output_expr.id
 
-        subject_args = get_args[1:]
-        if len(subject_args) == 1:
-            subject_constructor = subject_args[0]
-            if not isinstance(subject_constructor, ast.Call):
+        input_args = get_args[1:]
+        if len(input_args) == 1:
+            input_constructor = input_args[0]
+            if not isinstance(input_constructor, ast.Call):
                 raise parse_error(
-                    "Because you are using the shorthand form Get(ProductType, "
-                    "SubjectType(constructor args), the second argument should be a constructor "
+                    "Because you are using the shorthand form Get(OutputType, "
+                    "InputType(constructor args), the second argument should be a constructor "
                     "call, like `MergeDigest(...)` or `Process(...)`."
                 )
-            if not hasattr(subject_constructor.func, "id"):
+            if not hasattr(input_constructor.func, "id"):
                 raise parse_error(
-                    "Because you are using the shorthand form Get(ProductType, "
-                    "SubjectType(constructor args), the second argument should be a top-level "
+                    "Because you are using the shorthand form Get(OutputType, "
+                    "InputType(constructor args), the second argument should be a top-level "
                     "constructor function call, like `MergeDigest(...)` or `Process(...)`, rather "
                     "than a method call."
                 )
-            return product_type, subject_constructor.func.id  # type: ignore[attr-defined]
+            return output_type, input_constructor.func.id  # type: ignore[attr-defined]
 
-        subject_type, _ = subject_args
-        if not isinstance(subject_type, ast.Name):
+        input_type, _ = input_args
+        if not isinstance(input_type, ast.Name):
             raise parse_error(
-                "Because you are using the longhand form Get(ProductType, SubjectType, "
-                "subject_instance), the second argument should be a type, like `MergeDigests` or "
+                "Because you are using the longhand form Get(OutputType, InputType, "
+                "input), the second argument should be a type, like `MergeDigests` or "
```

Comment:
There's only one instance of this, but you could (if that text seemed stable enough) use a function to generate this message, which you could then invoke as a subroutine of the proposed `parse_get_shorthand_error()` method above.